impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java

```diff
@@ -0,0 +1,262 @@
+     *
+     * @param reader
+     * @return {@link PrivateKey}
+     * @throws IOException
+     */
+    PrivateKey getPrivateKeyFromPEM(Reader reader) throws IOException {
+        PrivateKey privateKey;
+
+        try (PEMParser pemParser = new PEMParser(reader)) {
+            JcaPEMKeyConverter jcaPEMKeyConverter = new JcaPEMKeyConverter();
+            Object pemContent = pemParser.readObject();
+
+            if (pemContent instanceof PEMKeyPair) {
+                PEMKeyPair pemKeyPair = (PEMKeyPair) pemContent;
+                KeyPair keyPair = jcaPEMKeyConverter.getKeyPair(pemKeyPair);
+                privateKey = keyPair.getPrivate();
+            } else if (pemContent instanceof PrivateKeyInfo) {
+                PrivateKeyInfo privateKeyInfo = (PrivateKeyInfo) pemContent;
+                privateKey = jcaPEMKeyConverter.getPrivateKey(privateKeyInfo);
+            } else {
+                throw new IllegalArgumentException("Unsupported Private Key format '" +
+                    pemContent.getClass().getSimpleName() + '"');
+            }
+        }
+
+        return privateKey;
+    }
+
+    /**
+     * Create token client config from the supplied API client config.
+     *
+     * Token client needs to retry http 401 errors only once which is not the case with the API client.
+     * We therefore effect this token client specific config by setting 'retryMaxElapsed'
+     * & 'retryMaxAttempts' fields.
+     *
+     * @param apiClientConfiguration
+     * @return ClientConfiguration to be used by token retrieval client.
+     */
+    ClientConfiguration constructTokenClientConfig(ClientConfiguration apiClientConfiguration) {
+        ClientConfiguration tokenClientConfiguration = new ClientConfiguration();
+
+        if (apiClientConfiguration.getClientCredentialsResolver() != null)
+            tokenClientConfiguration.setClientCredentialsResolver(apiClientConfiguration.getClientCredentialsResolver());
+
+        if (apiClientConfiguration.isCacheManagerEnabled())
+            tokenClientConfiguration.setCacheManagerEnabled(apiClientConfiguration.isCacheManagerEnabled());
+
+        tokenClientConfiguration.setCacheManagerTti(apiClientConfiguration.getCacheManagerTti());
+        tokenClientConfiguration.setCacheManagerTtl(apiClientConfiguration.getCacheManagerTtl());
+
+        if (apiClientConfiguration.getCacheManagerCaches() != null)
+            tokenClientConfiguration.setCacheManagerCaches(apiClientConfiguration.getCacheManagerCaches());
+
+        if (apiClientConfiguration.getRequestAuthenticatorFactory() != null)
+            tokenClientConfiguration.setRequestAuthenticatorFactory(apiClientConfiguration.getRequestAuthenticatorFactory());
+
+        if (apiClientConfiguration.getBaseUrlResolver() != null)
+            tokenClientConfiguration.setBaseUrlResolver(apiClientConfiguration.getBaseUrlResolver());
+
+        tokenClientConfiguration.setAuthenticationScheme(AuthenticationScheme.OAUTH2);
```

Comment:
The auth scheme for the token client _should_ be 'NONE' as there isn't any authentication available yet to use

Reply:
fixed